sdks/ruby/src/lib/hatchet/features/runs.rb

```diff
@@ -0,0 +1,423 @@
+              triggering_event_external_id: triggering_event_external_id
+            }
+          )
+        end
+
+        # Hack for uniqueness
+        run_id_to_run = {}
+        responses.each do |record|
+          record.rows.each do |run|
+            run_id_to_run[run.metadata.id] = run
+          end
+        end
+
+        run_id_to_run.values.sort_by(&:created_at).reverse
+      end
+
+      # List task runs according to a set of filters
+      #
+      # @param since [Time, nil] The start time for filtering task runs
+      # @param only_tasks [Boolean] Whether to only list task runs
+      # @param offset [Integer, nil] The offset for pagination
+      # @param limit [Integer, nil] The maximum number of task runs to return
+      # @param statuses [Array<HatchetSdkRest::V1TaskStatus>, nil] The statuses to filter task runs by
+      # @param until_time [Time, nil] The end time for filtering task runs
+      # @param additional_metadata [Hash<String, String>, nil] Additional metadata to filter task runs by
+      # @param workflow_ids [Array<String>, nil] The workflow IDs to filter task runs by
+      # @param worker_id [String, nil] The worker ID to filter task runs by
+      # @param parent_task_external_id [String, nil] The parent task external ID to filter task runs by
+      # @param triggering_event_external_id [String, nil] The event id that triggered the task run
+      # @return [HatchetSdkRest::V1TaskSummaryList] A list of task runs matching the specified filters
+      # @raise [Hatchet::Error] If the API request fails or returns an error
+      def list(
+        since: nil,
+        only_tasks: false,
+        offset: nil,
+        limit: nil,
+        statuses: nil,
+        until_time: nil,
+        additional_metadata: nil,
+        workflow_ids: nil,
+        worker_id: nil,
+        parent_task_external_id: nil,
+        triggering_event_external_id: nil
+      )
+        since = since || (Time.now - DEFAULT_SINCE_DAYS * 24 * 60 * 60)
+        until_time = until_time || Time.now
+
+        if (until_time - since) / (24 * 60 * 60) >= LARGE_DATE_RANGE_WARNING_DAYS
+          warn "Listing runs with a date range longer than #{LARGE_DATE_RANGE_WARNING_DAYS} days may result in performance issues. " +
+               "Consider using `list_with_pagination` instead."
+        end
+
+        @workflow_runs_api.v1_workflow_run_list(
+          @config.tenant_id,
+          since.utc.iso8601,
+          only_tasks,
+          {
+            offset: offset,
+            limit: limit,
+            statuses: statuses,
+            until: until_time.utc.iso8601,
+            additional_metadata: maybe_additional_metadata_to_kv(additional_metadata),
+            workflow_ids: workflow_ids,
+            worker_id: worker_id,
+            parent_task_external_id: parent_task_external_id,
+            triggering_event_external_id: triggering_event_external_id
+          }
+        )
+      end
```

Comment:
this should do pagination by default - it's copying the Python code, but `list_with_pagination` was added to avoid a breaking change. we can probably remove that method here and just have pagination happen automatically in the `list`